src/Bot.hs

```diff
@@ -38,26 +38,47 @@ app' = do
   post "/type" $ authorized $ requireParameter "text" typeRequest
 
 typeRequest :: Text -> TypeBot ()
-typeRequest = maybe badRequest slackRequest <=< hoogle
+typeRequest f = do
+  result <- hoogle f
+  maybe (noResultsSlack f) slackRequest result
 
 functionNameToUrl :: Text -> String
-functionNameToUrl x = unpack $ mconcat ["https://www.haskell.org/hoogle/?mode=json&hoogle=", x,"&start=1&count=1"]
+functionNameToUrl x = unpack $ mconcat [baseUrl, "?mode=json&hoogle=", x,"&start=1&count=1"]
+
+humanFriendlyUrl :: Text -> String
+humanFriendlyUrl x = unpack $ mconcat [baseUrl, "?hoogle=", x,"&start=1&count=1"]
+
+baseUrl :: Text
+baseUrl = "https://www.haskell.org/hoogle/"
 
 hoogle :: (MonadIO m) => Text -> m (Maybe SearchResult)
 hoogle f = liftIO $ do
   response <- simpleHttp . functionNameToUrl $ removeCommandChars f
   return $ firstResult =<< decode response
 
+noResultsSlack :: Text -> TypeBot ()
+noResultsSlack = slack . notFoundPayload
+
+notFoundPayload :: Text -> [String]
+notFoundPayload t = [ "{ \"text\":"
+                    , "\""
+                    , "I couldn't find a matching result, here's where I looked: "
+                    , humanFriendlyUrl $ removeCommandChars t
+                    , "\" }"
```

Comment:
I'll often use the `st` quasi-quoter as a sort of heredoc, because the dependency is already present in Yesod projects, but in this case it might even be worth pulling it in:

```hs
[st|{ "text": "... looked: #{humanFriendly Url $ removeCommandChars t}" }|]
```

https://hackage.haskell.org/package/shakespeare-text-0.11/docs/Text-Shakespeare-Text.html#v:st